lxd/storage/drivers/driver_zfs_utils.go

```diff
@@ -194,16 +194,60 @@ func (d *zfs) getDatasets(dataset string, types string) ([]string, error) {
 	return children, nil
 }
 
-func (d *zfs) setDatasetProperties(dataset string, options ...string) error {
+func (d *zfs) setDatasetProperties(dataset string, verifyCurrent bool, options ...string) error {
 	args := []string{"set"}
-	args = append(args, options...)
-	args = append(args, dataset)
 
-	_, err := shared.RunCommand("zfs", args...)
+	// Skip current property values verification.
+	if !verifyCurrent {
+		args = append(args, options...)
+		args = append(args, dataset)
+		_, err := shared.RunCommand("zfs", args...)
+		return err
+	}
+
+	keys := []string{}
+	values := []string{}
+
+	// Extract keys and values from options.
+	for _, option := range options {
+		property := strings.Split(option, "=")
+
+		if len(property) != 2 {
+			continue
+		}
+
+		keys = append(keys, property[0])
+
+		// Handle "posixacl" and "posix" equivalency to avoid duplicate events.
+		if property[0] == "acltype" && property[1] == "posixacl" {
```

Comment:
Can you please explain this a bit more? The question I am having is why can't we set `acltype=posixacl`?

Is it the same or should we instead update `zfsDefaultSettings` to use `posix` instead of `posixacl`?

Reply:
setting it to `posixacl` is the same as setting it to `posix` so without this check it actually will set it twice because the current value will be `posix` and the new will be `posixacl`. But in any case updating `zfsDefaultSettings` is probably a better approach. I will change this, thanks!